lib/IRGen/IRGenModule.cpp

```diff
@@ -2040,6 +2040,36 @@ bool IRGenModule::finalize() {
   }
   emitLazyPrivateDefinitions();
 
+  if (!getSILModule().getOptions().StopOptimizationAfterSerialization) {
```

Comment:
Could you instead define an optionset/enum with named values and use that, e.g.
```
enum ABIOptions: uint32_t {
  ObjectiveCInterop = 1 << 0;
  EmbeddedSwift = 1 << 1;
}

ABIValue |= ABIOptions::ObjectiveCInterop
ABIValue |= ABIOptions:: EmbeddedSwift
```

excuse my c++ syntax errors